de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/LemmaTechnologySubGenerator.xtend

```diff
@@ -0,0 +1,181 @@
+ *
+ * @author <a href="mailto:jonas.sorgalla@fh-dortmund.de">Jonas Sorgalla</a>
+ */
+class LemmaTechnologySubGenerator {
+    /** Factory to actually create and manipulate a LEMMA Technology */
+    val TECHNOLOGY_FACTORY = TechnologyFactory.eINSTANCE
+    /** Factory to actually create and manipulate a LEMMA DataModel */
+    val DATA_FACTORY = DataFactory.eINSTANCE
+    /** Predefined instance of the <strong>TechnologyModel</strong>. This instance is populated
+     * with the various technologies from the OpenAPI model, e.g., media types.
+     */
+    val myTechModel = TECHNOLOGY_FACTORY.createTechnology
+    /** OpenAPI schema which will be used as source for generation. */
+    OpenAPI openApi
+    /** SLF4j Logger */
+    val static logger = LoggerFactory.getLogger(LemmaServiceSubGenerator);
+    /** Location where the generated file is written */
+    String targetFolder
+    /** List of all unique media types. */
+    val mediaTypes = <String>newHashSet
+
+    new(OpenAPI api, String genPath, String techFilename) {
+        super()
+        logger.debug("Creating new Technology Sub Generator...")
+        this.openApi = api
+        this.targetFolder = genPath + techFilename
+    }
+
+    def Technology generate() {
+        logger.debug("Starting generation...")
+        logger.debug("Initializing model instance...")
+        initilize()
+        logger.debug("...Technology Model initialized!")
+        logger.debug("Creating protocols...")
+        createProtocols()
+        logger.debug("...Data Structures created!")
+
+        if (OpenApiUtil.writeModel(myTechModel, targetFolder)) {
+            logger.info("Technology model generation successful!")
+            logger.info('''Model written to «targetFolder»''')
+        } else
+            throw new Exception("Technology model generation failed :(")
+        return myTechModel
+    }
+
+    def createProtocols() {
+        //creates "rest" as the default openapi protocol
+        val rest = TECHNOLOGY_FACTORY.createProtocol
+        rest.name = "rest"
+        openApi.paths.forEach[path, pathSchema|searchMediaTimes(pathSchema)]
+        mediaTypes.forEach [ type |
+            val df = TECHNOLOGY_FACTORY.createDataFormat
+            df.formatName = type
+            rest.dataFormats.add(df)
+        ]
+        //Sets "application/json" as default format if the OpenAPI desc entails it at all.
+        // If not, the first encountered media type is set as default.
+        rest.defaultFormat = rest.dataFormats.findFirst[
+            it.formatName.equals("application/json")
+        ] ?: rest.dataFormats.get(0)
+        myTechModel.protocols.add(rest)
+    }
+
+    def searchMediaTimes(PathItem item) {
+        val httpVerbList = <Operation>newArrayList
+        httpVerbList.addAll(item.get, item.delete, item.put,
+            item.post, item.patch, item.head, item.options
+        )
+        httpVerbList.forEach [ verb |
+            if (verb !== null) {
```

Comment:
Use a `filter` to get rid of the `null` verbs before iterating over them to get rid of that crude `if` statement and the ugly indent:
```
httpVerbList.filter[it !== null].forEach[ verb | ...
```
Alternatively, you could implement a helper method to ensure that none of the entries in the verb list will ever be `null`:
```
def addNonNull(Set<Operation> verbSet, Operation verb) {
    if (verb !== null)
        verbSet.add(verb)
}

httpVerbSet.addNonNull(item.get)
httpVerbSet.addNonNull(item.delete)
...
```
You could, of course, also provide an `addAllNonNull` helper method that continuously invokes `addNonNull` but bear in mind than none of both methods actually preserves the protocol of the `Set` interface/`HashSet` implementation. It may therefore be even better to refer to a `Set` (or `List`) implementation that does proper `null`-checking (cf., e.g., https://stackoverflow.com/a/6437282) and then have a helper method that checks for the NPE.